lib/array_association_scope.rb

```diff
@@ -4,13 +4,41 @@ def self.scope(association, connection)
     INSTANCE.scope association, connection
   end
 
+  class BindSubstitution
+    def initialize(block)
+      @block = block
+    end
+
+    def bind_value(scope, column, value, alias_tracker)
+      substitute = alias_tracker.connection.substitute_at(column)
+      cast_type = ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Array
```

Comment:
AR assumes there be a `workflow.id = $1` comparison when loading the association, so the cast_type is for the workflow.id column which will be an integer. The bind param needs to be cast to an array of integers since we're doing a `workflow.id = ANY($1)` comparison.